server/runtime/src/main/java/org/infinispan/server/configuration/hotrod/HotRodServerConfigurationParser.java

```diff
@@ -123,6 +128,50 @@ private void parseHotRodConnector(XMLExtendedStreamReader reader, ConfigurationB
             }
          }
       }
+      if (implicitAuthentication) {
+         ServerSecurityRealm securityRealm = serverBuilder.endpoint().securityRealm();
+         builder.authentication().enable().securityRealm(securityRealm.getName());
+         String serverPrincipal = null;
+         for (KerberosSecurityFactoryConfiguration identity : securityRealm.getServerIdentities().kerberosConfigurations()) {
+            if (identity.getPrincipal().startsWith("hotrod/")) {
+               builder.authentication()
+                     .addMechanisms(SaslMechanismInformation.Names.GS2_KRB5, SaslMechanismInformation.Names.GSSAPI);
+               serverPrincipal = identity.getPrincipal();
+               break;
+            }
+            Server.log.debugf("Enabled Kerberos mechanisms for Hot Rod using principal '%s'", identity.getPrincipal());
+         }
+         if (securityRealm.hasFeature(ServerSecurityRealm.Feature.TOKEN)) {
+            builder.authentication().addMechanisms(SaslMechanismInformation.Names.OAUTHBEARER);
+            Server.log.debug("Enabled OAUTHBEARER mechanism for Hot Rod");
+         }
+         if (securityRealm.hasFeature(ServerSecurityRealm.Feature.TRUST)) {
+            builder.authentication().addMechanisms(SaslMechanismInformation.Names.EXTERNAL);
+            Server.log.debug("Enabled EXTERNAL mechanism for Hot Rod");
```

Comment:
may this Server log be used in the REST endpoints for tracing/debugging ? not a review request but a question for adding logs that actually lack IMO

Reply:
Only add logs that would be useful for helping in user scenarios. For internal aid, just use debugging. Also, in performance-critical areas debug/trace logs should be surrounded by `if(log.isxxxenabled())`